client/test/extension.test.ts

```diff
@@ -111,4 +120,65 @@ describe('Extension Tests', () => {
 
         treeSpy.should.have.been.called;
     });
+
+    it('should install native dependencies on first activation', async () => {
+        const dependencyManager = DependencyManager.instance();
+        mySandBox.stub(vscode.commands, 'registerCommand');
+        mySandBox.stub(dependencyManager, 'hasNativeDependenciesInstalled').returns(false);
+        const installStub = mySandBox.stub(dependencyManager, 'installNativeDependencies').resolves();
+        const tempRegistryExecuteStub = mySandBox.stub(TemporaryCommandRegistry.instance(), 'executeStoredCommands');
+
+        const context: vscode.ExtensionContext = ExtensionUtil.getExtensionContext();
+        await myExtension.activate(context);
+        installStub.should.have.been.called;
+        tempRegistryExecuteStub.should.have.been.called;
+    });
+
+    it('should not install native dependencies if already installed', async () => {
+        const dependencyManager = DependencyManager.instance();
+        mySandBox.stub(vscode.commands, 'registerCommand');
+        mySandBox.stub(dependencyManager, 'hasNativeDependenciesInstalled').returns(true);
+        const installStub = mySandBox.stub(dependencyManager, 'installNativeDependencies').resolves();
+
+        const context: vscode.ExtensionContext = ExtensionUtil.getExtensionContext();
+        await myExtension.activate(context);
+        installStub.should.not.have.been.called;
+    });
+
+    it('should handle any errors', async () => {

```

Comment:
Should this be more specific? i.e. "should handle any errors when installing dependencies"